### PR TITLE
Fix Select Sonar File No Overlap Case

### PR DESCRIPTION
**Pull request by ctuguinay**
- Fix `select_sonar_file` case where there is no overlap between region times and sonar file times
- Add a test to check for empty list output when there is no overlap
- Add a test to `parse_simrad_fname_time` to check if it works for another year and for a `.Sv` file

### CODE REVIEW

File: echoregions/regions2d/regions2d.py [hunk 1]
```diff
@@ -260,32 +260,44 @@ def select_sonar_file(
...
-        upper_idx = np.searchsorted(filetimes, times.max())
+        # Extract region time values
+        region_times = np.hstack(region["time"].values)
 
-        lower_idx = 0 if lower_idx < 0 else lower_idx
+        # Check if all sonar file times are completely below or above all region times
+        if np.all(sonar_file_times < region_times.min()) or np.all(
+            sonar_file_times > region_times.max()
+        ):
+            return []
```
Comment (ctuguinay) [inline echoregions/regions2d/regions2d.py:289]:
```suggestion
            print("Sonar file times did not overlap at all with region times. Returning empty list")
            return []
```

**Review by leewujung — APPROVED** [review]
LGTM!
